src/main/java/com/example/app/entity/Role.java

```diff
@@ -25,6 +25,6 @@
     @Column(name = "role_name", unique = true, nullable = false)
     private String roleName;
 
-    @ManyToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL, mappedBy = "roles")
-    private Collection<User> users;
+    @OneToMany(fetch = FetchType.EAGER, mappedBy = "role")
+    private Set<User> users = new HashSet<>(0);
```

Comment:
Interesting...do we really need this inline initialization here?

Reply:
Some cool guys from StackOverFlow said it's amazing solution!

Reply:
That's kinda strange...I'll double-check this now.

Reply:
Let's add TODO here and back to it later.